src/overlays/actors/ovl_Obj_Hamishi/z_obj_hamishi.c

```diff
@@ -30,17 +29,175 @@ const ActorInit Obj_Hamishi_InitVars = {
+
+        this->actor.posRot.pos.x =
+            this->actor.initPosRot.pos.x + (Math_Sins(this->shakePosPhase * 4) * this->shakePosSize);
+        this->actor.posRot.pos.z =
+            this->actor.initPosRot.pos.z + (Math_Coss(this->shakePosPhase * 7) * this->shakePosSize);
+        this->actor.shape.rot.x =
+            this->actor.initPosRot.rot.x + (s16)(Math_Sins(this->shakeRotPhase * 4) * this->shakeRotSize);
+        this->actor.shape.rot.z =
+            this->actor.initPosRot.rot.z + (s16)(Math_Coss(this->shakeRotPhase * 7) * this->shakeRotSize);
+    } else {
+        Math_ApproxF(&this->actor.posRot.pos.x, this->actor.initPosRot.pos.x, 1.0f);
+        Math_ApproxF(&this->actor.posRot.pos.z, this->actor.initPosRot.pos.z, 1.0f);
+        Math_ApproxUpdateScaledS(&this->actor.shape.rot.x, this->actor.initPosRot.rot.x, 0xBB8);
+        Math_ApproxUpdateScaledS(&this->actor.shape.rot.z, this->actor.initPosRot.rot.z, 0xBB8);
+    }
+}
+
+void ObjHamishi_Break(ObjHamishi* this, GlobalContext* globalCtx) {
+    s32 pad;
+    Vec3f velocity;
+    Vec3f pos;
+    s16 phi_s0 = 1000;
+    s16 gravity;
+    s16 phi_v0;
+    f32 temp_f20;
+    f32 temp_f22;
+    s32 i;
+
+    for (i = 0; i < ARRAY_COUNT(sEffectScales); i++) {
+        phi_s0 += 20000;
+
+        temp_f20 = Math_Rand_ZeroOne() * 10.0f;
+        pos.x = (Math_Sins(phi_s0) * temp_f20) + this->actor.posRot.pos.x;
+        pos.y = (Math_Rand_ZeroOne() * 40.0f) + this->actor.posRot.pos.y + 5.0f;
+        pos.z = (Math_Coss(phi_s0) * temp_f20) + this->actor.posRot.pos.z;
+
+        temp_f20 = (Math_Rand_ZeroOne() * 10.0f) + 2.0f;
+        velocity.x = Math_Sins(phi_s0) * temp_f20;
+        temp_f22 = Math_Rand_ZeroOne();
+        velocity.y = (Math_Rand_ZeroOne() * i * 2.5f) + (temp_f22 * 15.0f);
+        velocity.z = Math_Coss(phi_s0) * temp_f20;
+
+        if (i == 0) {
+            phi_v0 = 41;
+            gravity = -450;
+        } else if (i < 4) {
+            phi_v0 = 37;
+            gravity = -380;
+        } else {
+            phi_v0 = 69;
+            gravity = -320;
+        }
+
+        EffectSsKakera_Spawn(globalCtx, &pos, &velocity, &this->actor.posRot.pos, gravity, phi_v0, 30, 5, 0,
+                             sEffectScales[i], 3, 0, 70, 1, OBJECT_GAMEPLAY_FIELD_KEEP, &D_0500A5E8);
+    }
+
+    func_80033480(globalCtx, &this->actor.posRot, 140.0f, 6, 180, 90, 1);
+    func_80033480(globalCtx, &this->actor.posRot, 140.0f, 12, 80, 90, 1);
+}
+
+void ObjHamishi_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ObjHamishi* this = THIS;
+    s16 newRotY;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+
+    if (globalCtx->csCtx.state != 0) {
+        this->actor.uncullZoneForward += 1000.0f;
+    }
+    if (this->actor.shape.rot.y == 0) {
+        newRotY = Math_Rand_ZeroFloat(65536.0f);
+        this->actor.initPosRot.rot.y = newRotY;
+        this->actor.posRot.rot.y = newRotY;
+        this->actor.shape.rot.y = newRotY;
```

Comment:
You can probably avoid this temp bydoing something like this:
```suggestion
        this->actor.shape.rot.y =  this->actor.posRot.rot.y = this->actor.initPosRot.rot.y = Math_Rand_ZeroFloat(65536.0f);
```

Reply:
Good idea, I'll test if that matches and then commit it if so.